include/sparrow_v01/layout/list_layout/list_array.hpp

```diff
@@ -0,0 +1,144 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or mplied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "sparrow_v01/layout/array_base.hpp"
+#include "sparrow/layout/layout_iterator.hpp"
+#include "sparrow_v01/layout/nested_value_types.hpp"
+#include "sparrow/utils/iterator.hpp"
+#include "sparrow/utils/nullable.hpp"
+#include "sparrow/array/data_traits.hpp"
+#include "sparrow_v01/array_factory.hpp"
+#include "sparrow_v01/utils/memory.hpp"
+#include "sparrow_v01/utils/functor_index_iterator.hpp"
+#include "sparrow_v01/layout/layout_utils.hpp"
+
+namespace sparrow
+{
+    template <bool BIG>
+    class list_array_impl;
+
+    using list_array = list_array_impl<false>;
+    using big_list_array = list_array_impl<true>;
+
+    template <bool BIG>
+    struct array_inner_types<list_array_impl<BIG>> : array_inner_types_base
+    {
+        using array_type = list_array_impl<BIG>;
+        using inner_value_type = list_value2;
+        using inner_reference  = list_value2;
+        using inner_const_reference = list_value2;
+        using value_iterator = functor_index_iterator<detail::LayoutValueFunctor<array_type>>;
+        using const_value_iterator = functor_index_iterator<detail::LayoutValueFunctor<const array_type>>;
+        using iterator_tag = std::random_access_iterator_tag;
+    };
+
+    template <bool BIG>
+    class list_array_impl final : public array_base,
+                                  public array_crtp_base<list_array_impl<BIG>>
+    {
+        private:
```

Comment:
Formatting and homogeneity:
- private / public should be less indented
- the implementation should be after the class
- then a single private section would be enough